### PR TITLE
Blogpost: How to use Camel textual Route Debugger with Unit Test

### PR DESCRIPTION
**Pull request by apupier**

### CODE REVIEW

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.camel</groupId>
+                    <artifactId>camel-debug</artifactId>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
+<repositories>
```
Comment (essobedo) [inline content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:46]: this line can be removed

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                </property>
+            </activation>
+            <dependencies>
+                <dependency>
+                    <groupId>org.apache.camel</groupId>
+                    <artifactId>camel-debug</artifactId>
+                </dependency>
+            </dependencies>
+        </profile>
+    </profiles>
```
Comment (essobedo) [inline content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:45]:
```suggestion
    <profile>
        <id>camel.debug</id>
        <activation>
            <property>
                <name>camel.debug</name>
                <value>true</value>
            </property>
        </activation>
        <dependencies>
            <dependency>
                <groupId>org.apache.camel</groupId>
                <artifactId>camel-debug</artifactId>
            </dependency>
        </dependencies>
    </profile>
</profiles>
```

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+We also need to ensure that the test is starting after Debugger is attached and breakpoint set. For now, the only solution that I can propose is to insert a (ugly) `Thread.sleep` at the beginning of the test. It will work only if the test is triggering the route execution. If your route is automatically triggered like for a `timer`, some route executions might happen before the debugger is attached and breakpoints enabled.
+
+You should end up with something like that:
+
+```java
+class MainTest extends CamelMainTestSupport {
+	@Test
+	void myTest() throws Exception {
+		Thread.sleep(2000); // to let time to Debugger to attach and install breakpoints
```
Comment (essobedo) [inline content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:65]: Why not asking to launch the IDE in debug mode and to add breakpoint instead?

File: .gitignore [hunk 4]
```diff
@@ -9,7 +9,6 @@ resources
 .project
 .classpath
 .metadata
-.settings
```
Comment (essobedo) [inline .gitignore:12]: Not needed I guess?

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md (hunk 3 above)
Comment (apupier) [reply content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:65]: Wasn't able to have the Camel textual debugging working at the same time. But definitely something to improve.

File: .gitignore (hunk 4 above)
Comment (apupier) [reply .gitignore:12]: it is needed to have the VS Code task and launch configuration available in the repository

Comment (apupier) [reply .gitignore:12]: oh my bad, I was in another repository in my head. Effectively, not needed

**Comment by apupier** [comment]
I do not reproduce locally the error of PR job...

```
[check:html     ] public/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.html
[check:html     ]   28:49   error  Trailing whitespace  no-trailing-whitespace
[check:html     ]   48:283  error  Trailing whitespace  no-trailing-whitespace
```

**Review by davsclaus — APPROVED** [review]

**Comment by apupier** [comment]
Thanks for the reviews and the merge!

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+		}
+	]
+}
+```
+
+It allows to start the test with `camel-debug` on the classpath.
+
+## VSCode Launch configuration
+
+The following launch configuration must be created in the `.settings/launch.json`:
```
Comment (essobedo) [inline content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:114]: @apupier I'm wondering if the right path is rather `.vscode/launch.json`?

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+}
+```
+
+You might also want to introduce a longer timeout to let you time to debug.
+
+# IDE configuration
+
+## VS Code Task configuration
+
+The following VS Code task must be created in the `settings/tasks.json` file:
```
Comment (essobedo) [inline content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:86]: ditto

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md (hunk 5 above)
Comment (essobedo) [reply content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:114]: like in your fork https://github.com/apupier/camel-examples/tree/Blogpost-Example-Debug-Camel-In-Test/examples/main/.vscode

Comment (apupier) [reply content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:114]: created https://github.com/apache/camel-website/pull/863

File: content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md (hunk 6 above)
Comment (apupier) [reply content/blog/2022/06/HowToUseCamelRouteTextualDebuggerWithUnitTest/index.md:86]: created https://github.com/apache/camel-website/pull/863

**Pull request merged**